marimbabot_planning/src/lilypond_to_hitpoint.cpp

```diff
@@ -106,87 +107,110 @@ std::vector<std::tuple<geometry_msgs::PoseStamped, double, double>> lilypond_to_
-        marker.scale.z = 0.02;
-        marker.color.a = 1.0;
-        marker.color.r = 0.0;
-        marker.color.g = 1.0;
-        marker.color.b = 0.0;
-
-        // Add lifetime (duration of the note)
-        marker.lifetime = ros::Duration(std::get<2>(hits[i]));
-
-        // Publish the marker
-        hit_marker_pub.publish(marker);
-
-        // Check if this is not the last one
-        if(i < hits.size() - 1)
+        ROS_INFO("Converted lilypond sequence to cartesian poses and times");
+
+        // Loop over all hits and publish a marker at the correct time
+        double time = 0.0;
+
+        for(int i = 0; i < hits.size(); i++)
         {
-            // Get the time of the next hit
-            double next_time = std::get<1>(hits[i + 1]);
+            // Create a marker
+            visualization_msgs::Marker marker;
+            marker.header.frame_id = "base_link";
+            marker.header.stamp = ros::Time::now();
+            marker.ns = "marimba";
+            marker.id = 0;
+            // Show a downpointing gree arrow
+            marker.type = visualization_msgs::Marker::ARROW;
+            marker.action = visualization_msgs::Marker::ADD;
+            marker.pose.position = std::get<0>(hits[i]).pose.position;
+            // Offset the marker so it is not under the marimba
+            marker.pose.position.z += 0.1;
+            // Make it down pointing
+            marker.pose.orientation.x = 0.0;
+            marker.pose.orientation.y =  0.707;
+            marker.pose.orientation.z = 0.0;
+            marker.pose.orientation.w =  0.707;
+            marker.scale.x = 0.1;
+            marker.scale.y = 0.02;
+            marker.scale.z = 0.02;
+            marker.color.a = 1.0;
+            marker.color.r = 0.0;
+            marker.color.g = 1.0;
+            marker.color.b = 0.0;
+
+            // Add lifetime (duration of the note)
+            marker.lifetime = ros::Duration(std::get<2>(hits[i]));
+
+            // Publish the marker
+            hit_marker_pub.publish(marker);
+
+            // Check if this is not the last one
+            if(i < hits.size() - 1)
+            {
+                // Get the time of the next hit
+                double next_time = std::get<1>(hits[i + 1]);
 
-            // Sleep until the next hit
-            ros::Duration(next_time - time).sleep();
+                // Sleep until the next hit
+                ros::Duration(next_time - time).sleep();
 
-            // Update the time
-            time = next_time;
+                // Update the time
+                time = next_time;
+            }
         }
+
+        ros::Duration(0.1).sleep();
     }
 
-    ros::Duration(0.1).sleep();
+private:
+    ros::NodeHandle n_; 
+    ros::Publisher hit_marker_pub;
+    ros::Subscriber sub_;
+
+};
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "marimba_move_lilypond_input_viz");
+
+    // Create SubscribeAndPublish object
+    SubscribeAndPublish SAPObject;
```

Comment:
I would write this object name in snake case to be consistent with all the other code. Also at least from my understanding please don't write variables starting with upper case